r1cs/gadgets/std/src/sha256.rs

```diff
@@ -0,0 +1,392 @@
+//! Circuits for the [SHA-256] hash function and its internal compression
```

Comment:
Can we move this file in a separate module under r1cs/gadgets/crypto/prf ?

Reply:
Prf is not enabled now as a feature. Enabling it results in some errors. Should I choose a different folder to move the sha256.rs module?

Reply:
Mmm we should understand the error and fix it rather than choosing another folder.